### PR TITLE
coding CreateSalaryRequestChangeFromCheckpoint API

### PR DESCRIPTION
**Pull request by quangnguyenhongncc**
coding CreateSalaryRequestChangeFromCheckpoint API

## Summary by CodeRabbit

- **New Features**
	- Introduced a new method to create salary change requests from a checkpoint, enhancing the salary management process.
	- Added new data transfer objects (DTOs) to support the creation and management of salary change requests.

- **Bug Fixes**
	- Improved validation logic for creating and updating salary requests, ensuring better data integrity.

- **Refactor**
	- Updated existing methods for consistency and improved functionality in the salary request management process.

### CODE REVIEW

File: aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/Dto/CreateSalaryRequestDto.cs [hunk 1]
```diff
@@ -10,7 +11,7 @@
 namespace HRMv2.Manager.SalaryRequests.Dto
 {
     [AutoMapTo(typeof(SalaryChangeRequest))]
-    public class CreateSalaryRequestDto
+    public class CreateSalaryRequestDto : Entity<long>
```
Comment (tiennguyenhuuncc) [inline aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/Dto/CreateSalaryRequestDto.cs:14]: thêm : Entity<long> để làm gì thế?

File: aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/Dto/GetSalaryRequestFromCheckpointDto.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using static HRMv2.Constants.Enum.HRMEnum;
+
+namespace HRMv2.Manager.SalaryRequests.Dto
+{
+    public class GetSalaryRequestFromCheckpointDto : Entity<long>
```
Comment (tiennguyenhuuncc) [inline aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/Dto/GetSalaryRequestFromCheckpointDto.cs:14]: public class CreateSalaryChangeRequestFromCheckpointDto
    {
        public string Name { get; set; }
        public DateTime ApplyMonth { get; set; }
        public List<EmployeeSalaryChangeRequestDto> Employees { get ; set; }

    }

    public class EmployeeSalaryChangeRequestDto
    {
        public string EmailAddress { get; set; }
        public string ToLevelCode { get; set; }
        public double SalaryIncrease { get; set; }
    }

Comment (tiennguyenhuuncc) [reply aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/Dto/GetSalaryRequestFromCheckpointDto.cs:14]: update phần code còn lại theo input như trên

File: aspnet-core/src/HRMv2.Application/APIs/SalaryChangeRequests/SalaryChangeRequestAppService.cs [hunk 3]
```diff
@@ -150,5 +150,13 @@ public async Task<FileBase64Dto> GetTemplateToImportCheckpoint()
             return await _salaryRequestManager.GetTemplateToImportCheckpoint();
         }
 
+        [HttpPost]
+        [NccAuthentication]
+        public async Task<List<ResultSendChangeRequestDto>> CreateSalaryRequestChangeFromCheckpoint(GetSalaryRequestFromCheckpointDto input)
```
Comment (tiennguyenhuuncc) [inline aspnet-core/src/HRMv2.Application/APIs/SalaryChangeRequests/SalaryChangeRequestAppService.cs:155]: CreateSalaryRequestChangeFromCheckpoint
=> CreateSalaryChangeRequestFromCheckpointTool

File: aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/Dto/CreateSalaryRequestDto.cs (hunk 1 above)
Comment (quangnguyenhongncc) [reply aspnet-core/src/HRMv2.Core/Manager/SalaryRequests/Dto/CreateSalaryRequestDto.cs:14]: Để mình có thể lấy được Id của từng RequestChangeSalary ạ , lấy để có thể mở được EmployeeRequestChangeSalary ạ

**Pull request closed without merging**